brainframe/api/bf_codecs/condition_codecs.py

```diff
@@ -109,7 +108,7 @@ def from_dict(d: dict):
         with_attribute = None
         if d["with_attribute"] is not None:
             with_attribute = Attribute.from_dict(d["with_attribute"])
-        test = CountConditionTestType(d["test"])
+        test = ZoneAlarmCountCondition.TestType(d["test"])
```

Comment:
Something I've never been able to decide is if it's better to use `ZoneAlarmCountCondition.TestType` or `self.TestType`.

One prevents inheritance/overriding, and the other somewhat masks that they're class attributes

Reply:
You can't tell from this diff, but the method this is in is static, so I only have one option in this case. I guess I could turn it into a classmethod and use `cls`, but in general these `from_dict` methods are all static just kind of by convention.

Reply:
Obviously it'd be more work, but for the sake of discussion, what would the harm in making them classmethods if need be?

Reply:
I think it would be fine. I'm just not sure if I see a situation where someone would want to subclass and override TestType.